ACE/ace/Shared_Memory_Pool.cpp

```diff
@@ -222,19 +226,19 @@ ACE_Shared_Memory_Pool::handle_signal (int, siginfo_t *siginfo, ucontext_t *)
 ACE_Shared_Memory_Pool::ACE_Shared_Memory_Pool (
   const ACE_TCHAR *backing_store_name,
   const OPTIONS *options)
-  : base_addr_ (0),
-    file_perms_ (ACE_DEFAULT_FILE_PERMS),
+  : file_perms_ (ACE_DEFAULT_FILE_PERMS),
     max_segments_ (ACE_DEFAULT_MAX_SEGMENTS),
     minimum_bytes_ (0),
     segment_size_ (ACE_DEFAULT_SEGMENT_SIZE)
 {
   ACE_TRACE ("ACE_Shared_Memory_Pool::ACE_Shared_Memory_Pool");
 
-  // Only change the defaults if <options> != 0.
+  this->shm_addr_table_ = std::make_unique<void*[]>(this->max_segments_);
```

Comment:
The use of `shm_addr_table_[0]` and `shm_addr_table_[i]` for any `i > 0` are so different that it may be more clear just to keep `base_addr_` as it was and either have `[0]` be that same address or just keep the additional allocations in `shm_addr_table_`.